src/auth0/handlers/organizations.js

```diff
@@ -0,0 +1,208 @@
+        this.didCreate(data);
+        this.created += 1;
+      }).catch((err) => {
+        throw new Error(`Problem creating ${this.type} ${this.objString(item)}\n${err}`);
+      })
+    }).promise();
+  }
+
+  async updateOrganization(org, organizations) {
+    const { connections: existingConnections } = await organizations.find(orgToUpdate => orgToUpdate.name === org.name);
+
+    const params = { id: org.id };
+    const { connections } = org;
+
+    delete org.connections;
+    delete org.name;
+    delete org.id;
+
+    await this.client.organizations.update(params, org);
+
+    const connectionsToRemove = existingConnections.filter(c => !connections.find(x => x.connection_id === c.connection_id));
+    const connectionsToAdd = connections.filter(c => !existingConnections.find(x => x.connection_id === c.connection_id));
+    const connectionsToUpdate = connections.filter(c => existingConnections.find(x => x.connection_id === c.connection_id && x.assign_membership_on_login !== c.assign_membership_on_login));
+
+    // Handle updates first
+    await Promise.all(connectionsToUpdate.map(conn => this.client.organizations
+      .updateEnabledConnection(Object.assign({ connection_id: conn.connection_id }, params), { assign_membership_on_login: conn.assign_membership_on_login })
+      .catch(() => {
+        throw new Error(`Problem updating Enabled Connection ${conn.connection_id} for organizations ${params.id}`);
+      })));
+
+    await Promise.all(connectionsToAdd.map(conn => this.client.organizations
+      .addEnabledConnection(params, _.omit(conn, 'connection'))
+      .catch(() => {
+        throw new Error(`Problem adding Enabled Connection ${conn.connection_id} for organizations ${params.id}`);
+      })));
+
+    await Promise.all(connectionsToRemove.map(conn => this.client.organizations
+      .removeEnabledConnection(Object.assign({ connection_id: conn.connection_id }, params))
+      .catch(() => {
+        throw new Error(`Problem removing Enabled Connection ${conn.connection_id} for organizations ${params.id}`);
+      })));
+
+    return params;
+  }
+
+  async updateOrganizations(updates, orgs) {
+    await this.client.pool.addEachTask({
+      data: updates || [],
+      generator: item => this.updateOrganization(item, orgs).then((data) => {
+        this.didUpdate(data);
+        this.updated += 1;
+      }).catch((err) => {
+        throw new Error(`Problem updating ${this.type} ${this.objString(item)}\n${err}`);
+      })
+    }).promise();
+  }
+
+  async getType() {
+    if (this.existing) {
+      return this.existing;
+    }
+
+    if (!this.client.organizations || typeof this.client.organizations.getAll !== 'function') {
+      return [];
+    }
+
+    try {
+      const organizations = await this.getAllOrganizations();
+      for (let index = 0; index < organizations.length; index++) {
+        const connections = await this.client.organizations.connections.get({ id: organizations[index].id });
```

Comment:
We had a bug (#133) for roles and permissions where the client wasn't getting all permissions across all roles, it was only looking at the first page. If you change this to ` this.client.organizations.connections.getAll({ paginate: true, id: organizations[index].id });` then you should avoid a similar issue here.

@turcottedanny Can you please confirm that's the case?

Reply:
FYI - The limit is only 20 connections

Reply:
Ok cool! Assuming it will stay that way, then I think we're good here.

Reply:
I think it would be more future proof if we rely on pagination here, if we ever have to change the limit of connections or default value per page in node sdk, this might break. getAll also use the connection pool to prevent any rate limit problems.

Reply:
we wont' support more than 20 in the short term as it doesn't make sense